pkg/apis/kops/instancegroup.go

```diff
@@ -178,6 +181,16 @@ const (
 // SpotAllocationStrategies is a collection of supported strategies
 var SpotAllocationStrategies = []string{SpotAllocationStrategyLowestPrices, SpotAllocationStrategyDiversified, SpotAllocationStrategyCapacityOptimized}
 
+// InstanceMetadataOptions holds the fields required to configure instance metadata in an instance.
+// This currently holds fields to require EC2 Instance Metadata in EC2
+type InstanceMetadataOptions struct {
+	// HTTPPutResponseHopLimit is the max no of hops made by the PUT request which fetches the token(default '1')
+	HTTPPutResponseHopLimit *int64 `json:"httpPutResponseHopLimit,omitempty"`
+	// HTTPTokens can be either 'required' or 'optional'. If value is 'required', then EC2 IMDv2 will be enabled and
+	// EC2 IMDv1 will be disabled. If it is 'optional', both v1 and v2 will be enabled (default 'optional')
+	HTTPTokens *string `json:"httpTokens,omitempty"`
+}
+
```

Comment:
> The new API fields look good to me but I'll let others review as well.

@rifelpet in this case, I tend to disagree, I am not happy to ddd another structure. This should be flatten same as we do with spot options and other similar things.
I would suggest using.
```
MetadataOptionsHTTPPutResponseHopLimit
MetadataOptionsHTTPTokens
```

Reply:
That makes sense @hakman . will do so. Just to clear things out, do you want me to change the struct name to `MetadataOptions` from `InstanceMetadataOptions`?  I think `MetadataOptions` should be better on 2nd thought as cloudformation and terraform use this.

Reply:
on the 2nd hand. I am thinking whether `InstanceMetadataOptions` itself would be better as in cloudformation and terraform it is clear that the context refers to LaunchTemplates. The context may not be clear in InstanceGroup yaml is what I feel. Maybe we change the name to 
```
InstanceMetadataOptionsHTTPPutResponseHopLimit
InstanceMetadataOptionsHTTPTokens
```
Would love to know your thoughts on this @hakman :)

Reply:
OK, let's keep the `instance` part and remove the `options` part. The full name is way too long, at least for me.

Reply:
I'm curious why you prefer a more flattened structure? I would lean toward a nested structure because:
* Many of the existing InstanceGroupSpec fields use a nested structure (`MixedInstancesPolicy`, `IAM`, `RollingUpdate`) so being consistent would be more intuitive.
* The cloud provider SDKs all use a nested structure ([aws](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#ResponseLaunchTemplateData), [gcp](https://pkg.go.dev/google.golang.org/api/compute/v1#InstanceTemplate), [DO](https://pkg.go.dev/github.com/digitalocean/godo#DropletCreateRequest), [azure](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v48.2.0+incompatible/services/compute/mgmt/2020-06-30/compute#VirtualMachine), and [openstack](https://pkg.go.dev/github.com/gophercloud/gophercloud@v0.14.0/openstack/compute/v2/servers#Server) to an extent)
* a nested structure is less verbose for the user:
  ```yaml
  foo:
    bar: 1
    baz: 2
    qux: 3
  ```
  vs
  ```yaml
  fooBar: 1
  fooBaz: 2
  fooQux: 3
  ```
It would be trivial to flatten/unflatten the structure in a future API version too.

Reply:
Ah, forgot about the MIP one... let's leave as is then. Would be intuitive if all these AWS specific stuff would be under an AWS struct in the future.

Reply:
@hakman @rifelpet i shall be keeping the fields name as it is then.